bitcoin/src/lib.rs

```diff
@@ -13,21 +13,25 @@
 //! See README.md for detailed documentation about development and supported
 //! environments.
 //!
-//! ## Available feature flags
+//! # Features
 //!
-//! * `std` - the usual dependency on `std` (default).
-//! * `secp-recovery` - enables calculating public key from a signature and message.
-//! * `base64` - (dependency), enables encoding of PSBTs and message signatures.
-//! * `rand` - (dependency), makes it more convenient to generate random values.
-//! * `serde` - (dependency), implements `serde`-based serialization and
-//!                 deserialization.
+//! * `default` - enables `std` and `secp-recovery`.
+//! * `std` - the usual dependency on `std`.
+//! * `serde` - implements `serde`-based serialization and deserialization.
 //! * `secp-lowmemory` - optimizations for low-memory devices.
-//! * `bitcoinconsensus-std` - enables `std` in `bitcoinconsensus` and communicates it
-//!                            to this crate so it knows how to implement
-//!                            `std::error::Error`. At this time there's a hack to
-//!                            achieve the same without this feature but it could
-//!                            happen the implementations diverge one day.
-//! * `ordered` - (dependency), adds implementations of `ArbitraryOrdOrd` to some structs.
+//! * `secp-recovery` - enables calculating public key from a signature and message.
+//! * `bitcoinconsensus` - (dependency) enables validation of scripts/transactions by `rust-bitcoinconsensus`.
+//! * `bitcoinconsensus-std` - same as `bitcoinconsensus` but also enables `std`.
+//! * `rand` - makes it more convenient to generate random values.
+//! * `rand-std` - same as `rand` but also enables `std`.
```

Comment:
Do we really need this feature? We should document why.

Reply:
I think that users need it for `thread_rng` to work with the `rand` traits.

Reply:
Can't they just enable it in `rand` itself?

Reply:
Yes, if they add `rand` to their top-level Cargo.toml. But then they need to keep the versions in sync.

Reply:
If we were to propagate every feature where people would need to keep versions in sync we'd quickly end up with tons of features. I'd prefer to not do it and just wait until this gets resolved in Cargo. (Yeah, it'll take years, still not worth the clutter I think. And yes, I did see the proposal, I just don't remember where.)

Reply:
True, we can't do it for *every* feature. But I think doing it for `std` feature for all our depedencies is a manageable amount of feature-explosion.

Reply:
OK, I can accept this for the `std` feature.